javascript-constructors/student.js

```diff
@@ -0,0 +1,23 @@
+/* exported Student */
+function Student(firstName, lastName, subject) {
+  this.firstName = firstName;
+  this.lastName = lastName;
+  this.subject = subject;
+
+  this.getFullName = function () {
+    return this.firstName + ' ' + this.lastName;
+  };
+
+  this.getIntroduction = function () {
+    return 'Hello, my name is ' + this.getFullName() + ' and I am studying ' + this.subject + '.';
+  };
+}
+
+// Object.setPrototypeOf()
+Student.prototype.getFullName = function () {
+  return this.firstName + ' ' + this.lastName;
+};
+
+Student.prototype.getIntroduction = function () {
+  return 'Hello, my name is ' + this.getFullName() + ' and I am studying ' + this.subject + '.';
+};
```

Comment:
@devanrivera98 You should only define each of these methods once, on the prototype.